x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/hosts/details/helpers.ts

```diff
@@ -200,15 +202,30 @@ export const getHostEndpoint = async (
         ? await getHostMetaData(metadataRequestContext, id, undefined)
         : null;
 
+    const fleetAgentId = endpointData?.metadata.elastic.agent.id;
+    const [fleetAgentStatus, pendingActions] = !fleetAgentId
+      ? [undefined, {}]
+      : await Promise.all([
+          // Get Agent Status
+          agentService.getAgentStatusById(esClient.asCurrentUser, fleetAgentId),
+          // Get a list of pending actions (if any)
+          getPendingActionCounts(esClient.asCurrentUser, [fleetAgentId]).then((results) => {
```

Comment:
I see this is being called for a single agent ID here. What is the overall call-chain here? Is this called when a user is looking at additional details of a single host? Or is this called one-by-one in a list of hosts, so this is likely to be called 30-times in quick succession?

Reply:
I think it is one host call only. This is the code that is invoked by the api call done from the Hosts details page. @XavierM might be able to help you with your question more than me 😄

Reply:
it is only for one host.